### PR TITLE
Fix failing GitHub workflow tests

### PR DESCRIPTION
**Pull request by benank**
Disabled 4 tests that were failing 100% of the time on GitHub runners. Fixed temporary directory creation to use the runner's temporary directory. Also updated the workflow with newer versions for actions and split up large operations into multiple steps.

### CODE REVIEW

**Comment by benank** [comment]
Closing for now since tests no longer appear to be flaky. Re-open if issue persists.

**Comment by benank** [comment]
Reopening as this is still an issue.

**Comment by benank** [comment]
Closing as this is not occurring often.

**Comment by benank** [comment]
Reopening as this is happening much more frequently now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,28 +7,39 @@ on:
 jobs:
   codecov-python-39:
     runs-on: windows-latest
+    env:
+      PYTHONTRACEMALLOC: 1
```
Comment (kjohn-msft) [inline .github/workflows/ci.yml:11]: Only for debugging. Not shown to reduce test speedn.

File: src/core/tests/library/RuntimeCompositor.py [hunk 2]
```diff
@@ -45,6 +47,14 @@ def __init__(self, argv=Constants.DEFAULT_UNSPECIFIED_VALUE, legacy_mode=False,
         self.argv = argv if argv != Constants.DEFAULT_UNSPECIFIED_VALUE else ArgumentComposer().get_composed_arguments()
         self.vm_cloud_type = vm_cloud_type
         Constants.Paths.SYSTEMD_ROOT = os.getcwd() # mocking to pass a basic systemd check in Windows
+        self.is_github_runner = os.getenv('RUNNER_TEMP', None) is not None
```
Comment (kjohn-msft) [inline src/core/tests/library/RuntimeCompositor.py:50]: Github runners set this. Ben found this in a random issues/PR trying to fix things.

File: src/extension/tests/Test_ExtOutputStatusHandler.py [hunk 3]
```diff
@@ -72,6 +72,9 @@ def test_read_file(self):
         shutil.rmtree(dir_path)
 
     def test_update_file(self):
+        if self.runtime.is_github_runner:
+            return
+
```
Comment (kjohn-msft) [inline src/extension/tests/Test_ExtOutputStatusHandler.py:77]: Ignored on GitHub because of failures in reading and writing files.

**Review by kjohn-msft — APPROVED** [review]

**Pull request merged**